src/kbc_modules/eaa_kbc/mod.rs

```diff
@@ -0,0 +1,244 @@
+        let response: DecryptionResponse =
+            serde_json::from_str::<DecryptionResponse>(recv_string.as_str())?;
+
+        let payload_hashmap = match response.status.as_str() {
+            "OK" => response.data,
+            "Fail" => return Err(anyhow!(format!("Verdictd: {}", response.error.unwrap()))),
+            _ => return Err(anyhow!("Can't understand the DecryptionResponse status!")),
+        };
+
+        if let Some(hashmap_content) = payload_hashmap {
+            let encrypted_payload_string = base64::encode(&encrypted_payload);
+            let decrypted_payload_string = hashmap_content.get(&encrypted_payload_string).ok_or(anyhow!(
+                "There is no field matching the encrypted payload in the data field of DecryptionResponse"
+            ))?;
+            let decrypted_payload = base64::decode(decrypted_payload_string)?;
+            return Ok(decrypted_payload);
+        } else {
+            return Err(anyhow!(
+                "DecryptionResponse status is OK but the data is null!"
+            ));
+        }
+    }
+
+    /*
+    fn kbs_get_kek(&mut self, mut key_id_list: Vec<String>) -> Result<HashMap<String, String>> {
+        let mut result_kek_map: HashMap<String, String> = HashMap::new();
+        for index in 0..key_id_list.len() {
+            let kid = &key_id_list[index];
+            if let Some(cached_kek) = self.kek_cache.get(kid) {
+                result_kek_map.insert(kid.to_string(), cached_kek.to_string());
+                key_id_list.remove(index);
+            }
+        }
+        if key_id_list.is_empty() {
+            return Ok(result_kek_map);
+        }
+
+        let request = GetKekRequest {
+            command: String::from("Get KEK"),
+            kids: key_id_list.clone(),
+        };
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+        let recv_string: String = self.kbs_trans_and_recv(trans_data, "Get KEK")?;
+        let response: GetKekResponse =
+            serde_json::from_str::<GetKekResponse>(recv_string.as_str())?;
+        let kek_map = match response.status.as_str() {
+            "OK" => response.data,
+            "Fail" => return Err(anyhow!(format!("Verdictd: {}", response.error.unwrap()))),
+            _ => return Err(anyhow!("Can't understand the GetKekResponse status!")),
+        };
+        if kek_map.is_none() {
+            return Err(anyhow!("GetKekResponse status is OK but the key is null!"));
+        }
+
+        if let Some(hashmap_content) = kek_map {
+            for (kid, kek) in &hashmap_content {
+                result_kek_map.insert(kid.to_string(), kek.to_string());
+            }
+        } else {
+            return Err(anyhow!("GetKekResponse status is OK but the key is null!"));
+        }
+
+        Ok(result_kek_map)
+    }
+    */
+
+    fn kbs_trans_and_recv(&mut self, trans_data: &[u8], error_info: &str) -> Result<String> {
```

Comment:
Yes, the KBS protocol is based on TLS socket. We didn't add length indicators to this because the currently our single package is small, so we think it won't be splited during tcp transport.